src/reducer.ts

```diff
@@ -108,50 +139,120 @@ type ReducerCreator = {
+ * streams emits. Specifically, if the seed is a stream, this operator will not
+ * emit when the seed stream emits, but the first time an action of source
+ * stream emits after the seed stream has emitted. The seed stream will be
+ * unsubscribed after its first emit.
  *
  * @param seed The initial input to the first reducer call
  * @param reducers The reducer entries that should be combined
  */
 export const combineReducers = <State>(
-  seed: State,
+  seed: State | Observable<State>,
   reducers: RegisteredReducer<State, any>[]
 ): OperatorFunction<UnknownAction, State> => {
+  const actionReducers = reducers.filter(isActionReducer);
+  const streamReducers = reducers.filter(isStreamReducer);
   const reducersByActionType = new Map(
-    reducers.flatMap(reducerFn =>
+    actionReducers.flatMap(reducerFn =>
       reducerFn.trigger.actions.map(action => [action.type, reducerFn])
     )
   );
+
+  type ScanInput =
+    | readonly [typeof SEED_ORIGIN, State]
+    | readonly [typeof ACTION_ORIGIN, UnknownAction]
+    | readonly [number, any];
+  type InitializedAccumulator = {
+    initialized: true;
+    reduced: boolean;
+    state: State;
+  };
+  type Accumulator = { initialized: false } | InitializedAccumulator;
+
+  const seed$ = wrapInObservable(seed).pipe(
+    take(1),
+    map(payload => [SEED_ORIGIN, payload] as ScanInput)
+  );
+  const source$s = streamReducers.map((reducerFn, i) =>
+    reducerFn.trigger.source$.pipe(map(payload => [i, payload] as ScanInput))
+  );
+
   return pipe(
-    ofType(...reducers.flatMap(reducerFn => reducerFn.trigger.actions)),
-    scan((state, { type, payload }: UnknownAction) => {
-      const RegisteredReducer = reducersByActionType.get(type);
-      if (RegisteredReducer === undefined) {
-        // This shouldn't be possible
-        return state;
-      }
+    ofType(...actionReducers.flatMap(reducerFn => reducerFn.trigger.actions)),
+    map(action => [ACTION_ORIGIN, action] as ScanInput),
+    merge(seed$, ...source$s),
+    scan(
+      (accumulator, packet) => {
+        if (packet[0] === SEED_ORIGIN) {
+          return { initialized: true, reduced: false, state: packet[1] };
+        }
+        if (!accumulator.initialized) {
+          return accumulator;
+        }
+
+        if (packet[0] === ACTION_ORIGIN) {
+          const reducerFn = reducersByActionType.get(packet[1].type)!;
+          return {
+            initialized: true,
+            reduced: true,
+            state: reducerFn(accumulator.state, packet[1].payload),
+          };
+        }
 
-      return RegisteredReducer(state, payload);
-    }, seed)
+        const reducerFn = streamReducers[packet[0]];
+        return {
+          initialized: true,
+          reduced: true,
+          state: reducerFn(accumulator.state, packet[1]),
+        };
+      },
+      { initialized: false } as Accumulator
+    ),
+    filter(
+      (acc): acc is InitializedAccumulator => acc.initialized && acc.reduced
```

Comment:
Does this mean that the combine reducers stream doesn't emit until it's reduced once?

Shouldn't it also emit once it's been initialized?

Reply:
Yes, I would think the same?

Ok, I see, that correspond to how scan works.